File: April 2022/April_10.cpp
class Solution {
public:
    int calPoints(vector<string>& ops) {
        vector<int> arr;
        for(auto &op : ops){
            if(op == "+")
                arr.push_back(arr[arr.size() - 1] + arr[arr.size() - 2]);
            else if(op == "D")
                arr.push_back(2 * arr[arr.size() - 1]);
            else if(op == "C")
                arr.pop_back();
            else
                arr.push_back(stoi(op));
        }
        
        int ans = 0;
        for(auto &score : arr)
            ans += score;
        return ans;
    }
};

File: April 2022/April_10.java
class Solution {
    public int calPoints(String[] ops) {
        List<Integer> list = new ArrayList<>();
        for(String op : ops){
            if(op.equals("C"))
                list.remove(list.size() - 1);
            else if(op.equals("D"))
                list.add(2 * list.get(list.size() - 1));
            else if(op.equals("+"))
                list.add(list.get(list.size() - 1) + list.get(list.size() - 2));
            else
                list.add(Integer.parseInt(op));
        }
        int ans = 0;
        for(int i : list)
            ans += i;
        return ans;
    }
}
